node/chainSimulator/components/testOnlyProcessingNode.go

```diff
@@ -370,6 +372,11 @@ func (node *testOnlyProcessingNode) GetStatusCoreComponents() factory.StatusCore
 	return node.StatusCoreComponents
 }
 
+// GetIncomingHeaderHandler will return the incoming header handler
+func (node *testOnlyProcessingNode) GetIncomingHeaderHandler() process.IncomingHeaderSubscriber {
+	return node.IncomingHeaderHandler
+}
```

Comment:
Perhaps make this implementation follow the same pattern as `func (node *testOnlyProcessingNode) GetChainHandler() chainData.ChainHandler`